site/plugins/propaganda/propaganda.py

```diff
@@ -51,56 +50,61 @@ def _gen_html(self, destination, images_folder, conf):
                     'url': url.strip(),
-                    'image': os.path.join(item, 'logo_c.png')
+                    'image': os.path.join(item, conf['image_name'])
                 }
                 n += 1
-                items.append(list_item.format(**advertise))
-        conf["advertise_list"] = "\n".join(items)
+                items.append(list_item.format(**propaganda))
+        conf["propaganda_list"] = "\n".join(items)
 
         local_html = html
         for key, value in conf.items():
             local_html = local_html.replace("{{%s}}" % key, str(value))
 
-        with open(destination, "wb") as destination_file:
-            destination_file.write(local_html.encode('utf-8'))
-
+        with open(destination, "wt", encoding="utf8") as destination_file:
+            destination_file.write(local_html)
 
+    def _gen_html_task(self, images_folder, tasks, key):
+        """Generate task to generate html."""
+        destination = os.path.join(self.site.config['OUTPUT_FOLDER'], tasks[key]['generated_at'])
+        tasks[key]['key'] = key
+        task = {
+            'basename': self.name,
+            'name': destination,
+            'targets': [destination],
+            'actions': [(self._gen_html, [destination, images_folder, tasks[key]])],
+            'clean': True,
+            'uptodate': [utils.config_changed(tasks[key], 'nikola.plugins.task.propaganda'), destination]
+        }
+        return task
 
 
 
     def gen_tasks(self):
         """Generate tasks."""
-        conf = self.site.config['ADVERTISING']
 
-        destination = os.path.join(self.site.config['OUTPUT_FOLDER'], 'advertising.html')
+        conf = self.site.config['PROPAGANDA']
         images_folder = os.path.join(self.site.config['OUTPUT_FOLDER'], 'propaganda')
         filters = self.site.config['FILTERS']
-        conf["items"] = os.listdir(conf['source_folder'])
 
         # copy files to static site
-
         for task in utils.copy_tree(conf['source_folder'], images_folder, link_cutoff=images_folder):
             task['basename'] = self.name
-            task['uptodate'] = [utils.config_changed(conf, 'nikola.plugins.task.advertising')]
+            task['uptodate'] = [utils.config_changed(conf, 'nikola.plugins.task.propaganda')]
             yield utils.apply_filters(task, filters, skip_ext=['.html'])
 
+        conf["items"] = os.listdir(conf['source_folder'])
+
         # generate html to insert into sidebar
 
-        task = {
-            'basename': self.name,
-            'name': destination,
-            'targets': [destination],
-            'actions': [(self._gen_html, [destination, images_folder, conf])],
-            'clean': True,
-            'uptodate': [utils.config_changed(conf, 'nikola.plugins.task.advertising')]
-        }
+        for key in conf["tasks"].keys():
+            task = self._gen_html_task(images_folder, conf["tasks"], key)
+            yield utils.apply_filters(task, filters)
```

Comment:
Why not iterate on .items() and pass to _gen_html_task directly the key and the task?